xcm/xcm-emulator/example/src/lib.rs

```diff
@@ -0,0 +1,438 @@
+// Copyright 2023 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use frame_support::{pallet_prelude::Weight, traits::GenesisBuild};
+use sp_runtime::AccountId32;
+
+use xcm_emulator::{decl_test_network, decl_test_parachain, decl_test_relay_chain};
+
+decl_test_relay_chain! {
+	pub struct KusamaNet {
```

Comment:
```suggestion
	pub struct KusamaRelay {
```

Reply:
Why is this even called "kusama"? It does not seem to be reading in any actual network state. In fact, it's setting the default XCM version to 3 when it is really 2 on Kusama. So this is clearly _not_ Kusama nor an accurate representation of it.

Reply:
It's called Kusama, because it's using the Kusama runtime definition. Note that those are just the examples that came with xcm-emulator in the original repo, unchanged. Those are meant to be used as guidance for builders in terms of structure and code, nothing more. 

XCM-Emulator is not designed to be reading the chain state yet, it simply allows for using pre-defined runtime definitions to test message passing between them. I'm sure this can be further enhanced to do more advanced testing, however this PR's aim is to move XCM-Emulator into the Cumulus repo and making sure it compiles/runs with the current versions of Cumulus and Polkadot. 

Furthermore, we are setting Kusama's XCM version to `3` in [parachains-integration-tests](https://github.com/paritytech/parachains-integration-tests/tree/master/examples/statemine), those are the tests that are used for release checks, so that is consistent with what we are doing here.